packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1372,28 +1411,7 @@ export class MergeTree {
 			// Perform slides after all segments have been acked, so that
 			// positions after slide are final
 			if (opArgs.op.type === MergeTreeDeltaType.REMOVE) {
-				// if the slide destination is further, then we have to slide
-				// references in reverse to preserve their order
-				const newSegment = this._getSlideToSegment(pendingSegmentGroup.segments[0]);
-				const shouldReverse =
-					pendingSegmentGroup.segments[0] &&
-					newSegment &&
-					pendingSegmentGroup.segments[0].ordinal < newSegment.ordinal;
-
-				let idx = shouldReverse ? pendingSegmentGroup.segments.length - 1 : 0;
-
-				while (idx >= 0 && idx < pendingSegmentGroup.segments.length) {
-					const pendingSegment = pendingSegmentGroup.segments[idx];
-					if (!overlappingRemoves[idx]) {
-						this.slideAckedRemovedSegmentReferences(pendingSegment);
-					}
-
-					if (shouldReverse) {
-						idx--;
-					} else {
-						idx++;
-					}
-				}
+				this.slideAckedRemovedSegmentReferences(pendingSegmentGroup.segments);
```

Comment:
i'm literally looking at the same bug right now 😆 I also might be related to the contiguous issue @scarlettjlee brought up.

Reply:
Thanks for tagging me. It's not related to my issue of contiguousness but is another issue that I hadn't thought about.  When I slide them back, I can lose the order. I could preserve the relative order of the slid references, although that can result in two change ops on the same interval if the start and end of an interval have another interval endpoint in between.

Reply:
Does preserving order only matter for local refs used by revertible or all local refs?

Reply:
intervals don't care about it. Some arbitrary usage may or may not care... but preserving the order on slide isn't less efficient (at least asymptotically) than not doing so, so we might as well do it for all refs.